### ISSUE TITLE
Variant annotation in VCF near-term concordance

### ISSUE THREAD

**Issue by stevenbrenner** [ga4gh/schemas#216]
The GA4GH Variant Annotation task teamis focused on providing a schema for reporting annotations on variants consistent with the broader GA4GH APIs.  This project will involve a conceptual study first.  See: https://github.com/ga4gh/schemas/issues/214

As an immediate near-term task, the GA4GH annotation task team will adopt and provide feedback on existing extensive efforts by Pablo Cingolani, Fiona Cunningham, Will McLaren & Kai Wang to harmonize the current variant annotations in VCF format.  See:
https://docs.google.com/document/d/1kjr7vpMjGp0DT3MbInsNIZOnopna3uN93ufSEF0wRuM/edit?pli=1

This thread is to capture discussion about this format.

**Comment by selewis** [comment]
It might be useful to also look at GVF (Genome Variation Format). Not
necessarily to use it directly as is (but could be another possibility),
but to aid in developing a complete final spec. (
http://www.sequenceontology.org/resources/gvf.html).

Also if this format is to be broadly adopted (i.e. not just for human),
then this statement "When comparing genomic coordinates, the comparison
should be done first by chromosome names (compared alphabetically), then by
start position, and finally by end position." A lot of organisms are still
dealing with fragmentary genomes and the chromosome name won't be known.
Wouldn't the ID of reference genome sequence be more flexible.

Concur with Reece on specifically stating what kind of annotations you're
speaking of. It's clear in the doc that these are solely annotations of
variant-type, but it doesn't jump right out first thing.

-S

**Comment by davecap** [comment]
This thread seems to be somewhat inactive, but has anyone thought about base64url encoding a JSON blob into the VCF INFO field? This would allow embedding annotations in a standardized GA4GH schema into VCFs, and would substitute the error-prone parsing of long and complex INFO fields with a simple, two-step de-serialization process supported by all languages.

Of course, it removes the possibility of an INFO field being humanly readable... but do people parse them by eye?

**Comment by pgrosu** [comment]
@davecap, it's a interesting idea but we would like to decouple data from annotations so that they can be updated independently.

**Comment by reece** [comment]
I agree with @pgrosu -- separating observations (in VCF) from annotations and interpretations (elsewhere) leads more naturally to a better conceptual data model (particularly multiple interpretations for a given observation).

**Comment by fnothaft** [comment]
@reece agreed as well; the separated genotypes/variant annotations model is the data model we use in [ADAM](https://www.github.com/bigdatagenomics/bdg-formats).
